tests/nat/middleware/test_dynamic_middleware.py

```
@@ -0,0 +1,503 @@
+# ==================== Middleware Invoke/Stream Tests ====================
+
+
+async def test_middleware_invoke_calls_next_with_no_policies(mock_builder):
+    """Test that invoke delegates to call_next when no policies are configured."""
+    config = DynamicMiddlewareConfig(register_workflow_functions=False)
+    middleware = DynamicFunctionMiddleware(config=config, builder=mock_builder)
+
+    test_input = {"value": "test"}
+    expected_output = {"result": "success"}
+
+    async def mock_call_next(*args, **kwargs):
```

Ruff (0.14.8)

Unused function argument: `kwargs`

(ARG001)